src/network/connectionthreads.cpp

```diff
@@ -956,8 +956,8 @@ void ConnectionReceiveThread::receive(SharedBuffer<u8> &packetdata,
 			// command was sent reliably.
 		}
 
-		/* The peer was not found in our lists. Add it. */
-		if (peer_id == PEER_ID_INEXISTENT) {
+		/* The peer was not found in our lists. Add it (unless we are the client). */
+		if (peer_id == PEER_ID_INEXISTENT && !m_connection->ConnectedToServer()) {
```

Comment:
it should return directly rather than doing this

Reply:
Done.